### PR TITLE
ref(endpoints): Relay directory

### PR DESCRIPTION
**Pull request by mgaeta**
In Backend TSC, we decided that it's time to start organizing some of our larger, flat directories. In this PR I move Relay-related endpoint files into a directory.

### CODE REVIEW

**Review by ceorourke — COMMENTED** [review]
Looks good to me, is our team actually the codeowner for relay stuff though?

**Comment by mgaeta** [comment]
There aren't any CODEOWNERS rules for `src/sentry/api/endpoints/relay*`.

File: src/sentry/api/endpoints/relay/project_configs.py [hunk 1]
```diff
@@ -26,6 +26,7 @@ def _sample_apm():
 class RelayProjectConfigsEndpoint(Endpoint):
     authentication_classes = (RelayAuthentication,)
     permission_classes = (RelayPermission,)
+    enforce_rate_limit = False
```
Comment (jjbayer) [inline src/sentry/api/endpoints/relay/project_configs.py:29]: Was this endpoint ever rate limited? Or was there some other logic in place that prevented rate limiting?

**Review by jjbayer — APPROVED** [review]

**Comment by jan-auer** [comment]
Let's also assign `@getsentry/owners-ingest` as codeowner of this, please, and make sure that all ownership rules in Sentry are updated (if there are any).

File: src/sentry/api/endpoints/relay/project_configs.py (hunk 1 above)
Comment (jan-auer) [reply src/sentry/api/endpoints/relay/project_configs.py:29]: In the past it was not, until general rate limits were introduced for the APIs. This change follows a discussion I've had with @AniketDas-Tekky last week, deciding that we'll disable all rate limits on endpoints that are called exclusively by processing Relays.

Comment (untitaker) [reply src/sentry/api/endpoints/relay/project_configs.py:29]: thanks for catching this @jan-auer, I was aware that we were introducing general rate limits but didn't think about impact to relay apis

**Pull request merged**